### DISCUSSION TITLE
Updating Bonsai scripts - problem with ExpressionTransform and PythonTransform

### DISCUSSION THREAD

**Discussion by liadJB** [bonsai-rx/bonsai#1691] [Q&A] [answered]
Hi, 

I am trying to update to the latest Bonsai version. However, I have got some problems with previous scripts not updating correctly. 
In both cases, the problem is with ExpressionTransform and PythonTransform that are greyed out and cannot operate the script.
I assume with ExpressionTransform I can just change each and every script manually and then copy paste the internal code in a text editor, but I will have to do it in about 30 different scripts, so I wondered if there is an automatic workaround?
With PythonTransform I couldn't find the node, is it in a different non-default package now?

An example script before and after:
[Update.zip](https://github.com/bonsai-rx/bonsai/files/14433099/Update.zip)

many thanks

Liad

**Comment by bruno-f-cruz** [comment]
Save for the daqmx operator (not sure what package you are using) the following enviornment is able to compile your `before` workflow.2
[Bonsai.config.txt](https://github.com/bonsai-rx/bonsai/files/14450045/Bonsai.config.txt)

**Reply by liadJB** [reply]
Great thanks.
I still have a few problems, but it seems after those it should be all set with the new version:

It seems that for some reason the video visualiser is gone. Now I am getting only textVisualiser on the Image node. Is there any thing I should specifically install or change?

When I used LogEvent in the past with a Selector: Ori,SFreq,TFreq,Contrast the log wrote all the selected value names too, e.g., Ori=180, SFreq=20...'. It seems like this not the case anymore. Is that correct? Do I now have to write in the formatter : "Ori={0}, SFreq = {1} etc?

Thanks for your help,

Liad

**Reply by liadJB** [reply]
It now completely stopped working. 
After installing Bonsai Visualizer package, bonsai started crashing.
I uninstalled it and installed it again. But this time it crashes mid installation from the config file and crashes constantly on a FileNotFound Error. See attached image
![image](https://github.com/bonsai-rx/bonsai/assets/48135253/12fa28ad-c6b4-4d4a-a90a-d5b12e2d19ca)
Installing the main package, uninstalling it, or trying to redeploy the package do not help. Our experiments do not work now as the same thing happens when I try to revert to the older version. I would really appreciate some help

Liad

**Reply by bruno-f-cruz** [reply]
Delete all your packages by deleting the package folder before running the bonsai executable. Second, What version of bonsai are you trying to install? Make sure the executable is compatible with the version you have in the config file. Can you add your most recent bonsai.config file?

**Comment by liadJB** [comment]
The latest one from the website. I am using the config file you sent me.
Is there a mismatch?
It happens also when I install directly from the installer though
Could you maybe tell me all the possible caches and folders that might
somehow interfere with the installation? I can delete those and try again
Thanks

**Reply by liadJB** [reply]
Update:  When I went to releases and used Bonsai 2.8.1 everything worked fine.
The main version from the website even with a clean install, crashed. 

If I try to install Visualizers or anything extra to be able to see a video live causes Bonsai to crash.

I am currently on version 2.8.1 with the caveats above: I cannot install visualizers to be able to view videos and the log operator functions differently. 

I would really appreciate any solution for the above, since we now seem to not be able to roll back to version 2.6.3

Thanks a lot!

**Reply by bruno-f-cruz** [reply, marked as answer]
The environment I sent using the latest Bonsai version bootstraps just fine on my computer. Can you give the attached zip a try? I have also added the starter package to the zip which already installs all the visualizers you need. Please make sure to read https://bonsai-rx.org/docs/articles/environments.html and version control your bonsai environment.



[env.zip](https://github.com/bonsai-rx/bonsai/files/14776467/env.zip)

**Reply by liadJB** [reply]
Hi, what you sent indeed worked well. I wonder, can you think of the reason the main installation or the portable version with the previous config files did not work? (Just for future reference)

**Reply by bruno-f-cruz** [reply]
You should probably try to reinstall everything from scratch, I suspect you have some remnants of some package or config file that you did not remove. Not sure...
